### PR TITLE
Move remove all issues

### LINKED ISSUE #877: Make removeAllIssues in Logger protected
Should we make the removeAllIssues method protected in the Logger class.  Importer flattenModel is one place where it isn't called unlike parseModel, and validateModel.

### PR DESCRIPTION
**Pull request by hsorby**
Fixes #877.

### CODE REVIEW

**Comment by hsorby** [comment]
Based on #891.

**Review by agarny — DISMISSED** [review]

File: src/importer.cpp [hunk 1]
```diff
@@ -621,6 +621,7 @@ bool Importer::resolveImports(ModelPtr &model, const std::string &basePath)
     bool status = true;
     History history;
 
+    pFunc()->removeAllIssues();
```
Comment (nickerso) [inline src/importer.cpp:624]: `resolveImport` documentation should be updated to note this behaviour

File: src/importer.cpp [hunk 2]
```diff
@@ -777,6 +778,7 @@ void flattenComponentTree(const ComponentEntityPtr &parent, ComponentPtr &compon
 
 ModelPtr Importer::flattenModel(const ModelPtr &model)
 {
+    pFunc()->removeAllIssues();
```
Comment (nickerso) [inline src/importer.cpp:781]: `flattenModel` documentation should be updated to reflect this behaviour.

**Review by nickerso — CHANGES_REQUESTED** [review]

File: src/importer.cpp (hunk 1 above)
Comment (hsorby) [reply src/importer.cpp:624]: 0d138f4

File: src/importer.cpp (hunk 2 above)
Comment (hsorby) [reply src/importer.cpp:781]: 0d138f4

**Review by agarny — APPROVED** [review]

**Review by nickerso — APPROVED** [review]

**Pull request merged**